server/sandbox_run_linux.go

```diff
@@ -385,6 +388,28 @@ func (s *Server) runPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 	}
 	g.AddAnnotation(annotations.CgroupParent, cgroupParent)
 
+	// check memory limit is greater than the minimum memory limit of 4Mb
+	// expand the cgroup slice path
+	slicePath, err := systemd.ExpandSlice(cgroupParent)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error expanding systemd slice path for %q", cgroupParent)
+	}
+	// read in the memory limit from the memory.limit_in_bytes file
+	fileData, err := ioutil.ReadFile(filepath.Join(cgroupMemoryPath, slicePath, "memory.limit_in_bytes"))
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading memory.limit_in_bytes file for slice %q", cgroupParent)
+	}
+	// strip off the newline character and convert it to an int
+	strMemory := string(fileData[:len(fileData)-1])
```

Comment:
If fileData is empty, but exists, does ReadFile succeed? if so, this will segfault. super odd case, but could save a headache later

Reply:
I am pretty sure it works fine and you end up with an empty byte array, but will look into it further.

Reply:
sorry I wasn't clear, that's what I'm worried about. wouldn't fileData[:len(fileData)-1] try to make an array with fileData[:-1] and then segfault?

Reply:
got it, fixed now. Using strings.TrimRight instead